velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -36,68 +34,93 @@ namespace fs = std::filesystem;
+    facebook::velox::parse::registerTypeResolver();
     filesystems::registerLocalFileSystem();
     parquet::registerParquetReaderFactory();
+    tempDirectory_ = exec::test::TempDirectoryPath::create();
+
     auto hiveConnector =
         connector::getConnectorFactory(
             connector::hive::HiveConnectorFactory::kHiveConnectorName)
             ->newConnector(kHiveConnectorId, nullptr);
     connector::registerConnector(hiveConnector);
-    tempDirectory_ = exec::test::TempDirectoryPath::create();
-    lineitemType_ =
-        ROW({"orderkey",
-             "partkey",
-             "suppkey",
-             "linenumber",
-             "quantity",
-             "extendedprice",
-             "discount",
-             "tax",
-             "returnflag",
-             "linestatus",
-             "shipdate",
-             "commitdate",
-             "receiptdate",
-             "shipinstruct",
-             "shipmode",
-             "comment"},
-            {BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             BIGINT(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             DOUBLE(),
-             VARCHAR(),
-             VARCHAR(),
-             DATE(),
-             DATE(),
-             DATE(),
-             VARCHAR(),
-             VARCHAR(),
-             VARCHAR()});
+
+    // Convert decimal columns to double.
+    const std::string lineitemSQL = R"(
+    COPY (SELECT l_orderkey, l_partkey, l_suppkey, l_linenumber, l_quantity,
+    l_extendedprice::DOUBLE as l_extendedprice,
+    l_discount::DOUBLE as l_discount, l_tax::DOUBLE as l_tax,
+    l_returnflag, l_linestatus, l_shipdate, l_receiptdate,
+    l_shipinstruct, l_shipmode, l_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["lineitem"] =
+        writeTableToParquet("lineitem", lineitemSQL);
+
+    const std::string ordersSQL = R"(
+    COPY (SELECT o_orderkey, o_custkey, o_orderstatus,
+    o_totalprice::DOUBLE as o_totalprice,
+    o_orderdate, o_orderpriority, o_clerk, o_shippriority, o_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["orders"] = writeTableToParquet("orders", ordersSQL);
+
+    const std::string customerSQL = R"(
+    COPY (SELECT c_custkey, c_name, c_address, c_nationkey, c_phone,
+    c_acctbal::DOUBLE as c_acctbal, c_mktsegment, c_comment
+    FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})
+    )";
+    parquetFilesPaths_["customer"] =
+        writeTableToParquet("customer", customerSQL);
+
+    planNodeIdGenerator_ = std::make_shared<PlanNodeIdGenerator>();
   }
 
-  void TearDown() override {
+  static void TearDownTestSuite() {
     connector::unregisterConnector(kHiveConnectorId);
     parquet::unregisterParquetReaderFactory();
   }
 
+  /// Write the DuckDB TPC-H table to a Parquet file and return the file path
+  ///
+  /// - tableName is name of table from TPC-H tables
```

Comment:
Use @param to document parameters.

Reply:
@9prady9 Apologies for asking you to remove `@param` earlier. I went through https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md
and see that it does mention 
```
For multi-line comments:
   Velox will follow the doxygen comment style. 
```